plugin/grpctrace/interceptor_test.go

```diff
@@ -15,127 +15,291 @@ package grpctrace
+func TestUnaryClientInterceptor(t *testing.T) {
+	exp := &testExporter{spanMap: make(map[string][]*export.SpanData)}
+	tp, _ := sdktrace.NewProvider(sdktrace.WithSyncer(exp),
+		sdktrace.WithConfig(sdktrace.Config{
+			DefaultSampler: sdktrace.AlwaysSample(),
+		},
+		))
+
+	clientConn, err := grpc.Dial("fake:connection", grpc.WithInsecure())
+	if err != nil {
+		t.Fatalf("failed to create client connection: %v", err)
+	}
+
+	tracer := tp.Tracer("grpctrace/client")
+	unaryInterceptor := UnaryClientInterceptor(tracer)
+
+	req := &mockProtoMessage{}
+	reply := &mockProtoMessage{}
+	uniInterceptorInvoker := &mockUICInvoker{}
+
+	checks := []struct {
+		name         string
+		expectedAttr map[core.Key]core.Value
+		eventsAttr   []map[core.Key]core.Value
+	}{
 		{
-			"FullyQualifiedMethodName",
-			"serviceName",
-			"/github.com.foo.%s/bar",
+			name: "/github.com.serviceName/bar",
+			expectedAttr: map[core.Key]core.Value{
+				rpcServiceKey:  core.String("serviceName"),
+				netPeerIPKey:   core.String("fake"),
+				netPeerPortKey: core.String("connection"),
+			},
+			eventsAttr: []map[core.Key]core.Value{
+				{
+					messageTypeKey: core.String("SENT"),
+					messageIDKey:   core.Int(1),
+				},
+				{
+					messageTypeKey: core.String("RECEIVED"),
+					messageIDKey:   core.Int(1),
+				},
+			},
 		},
 		{
-			"SimpleMethodName",
-			"serviceName",
-			"/%s/bar",
+			name: "/serviceName/bar",
+			expectedAttr: map[core.Key]core.Value{
+				rpcServiceKey: core.String("serviceName"),
+			},
+			eventsAttr: []map[core.Key]core.Value{
+				{
+					messageTypeKey: core.String("SENT"),
+					messageIDKey:   core.Int(1),
+				},
+				{
+					messageTypeKey: core.String("RECEIVED"),
+					messageIDKey:   core.Int(1),
+				},
+			},
 		},
 		{
-			"MethodNameWithoutFullPath",
-			"serviceName",
-			"%s/bar",
+			name:         "serviceName/bar",
+			expectedAttr: map[core.Key]core.Value{rpcServiceKey: core.String("serviceName")},
 		},
 		{
-			"InvalidMethodName",
-			"",
-			"invalidName",
+			name:         "invalidName",
+			expectedAttr: map[core.Key]core.Value{rpcServiceKey: core.String("")},
 		},
 		{
-			"NonAlphanumericMethodName",
-			"serviceName_123",
-			"/github.com.foo.%s/method",
+			name:         "/github.com.foo.serviceName_123/method",
+			expectedAttr: map[core.Key]core.Value{rpcServiceKey: core.String("serviceName_123")},
 		},
 	}
 
-	for _, tc := range testCases {
-		t.Run(tc.testName, tc.testUCISetsExpectedNameAttribute)
+	for _, check := range checks {
+		err = unaryInterceptor(context.Background(), check.name, req, reply, clientConn, uniInterceptorInvoker.invoker)
+		if err != nil {
+			t.Fatalf("failed to run unary interceptor: %v", err)
+		}
+
+		spanData, ok := exp.spanMap[check.name]
+		if !ok || len(spanData) == 0 {
+			t.Fatalf("no span data found for name < %s >", check.name)
+		}
+
+		attrs := spanData[0].Attributes
+		for _, attr := range attrs {
+			expectedAttr, ok := check.expectedAttr[attr.Key]
+			if ok {
```

Comment:
If `attr.Key` is not found in the expected attributes an error should be raised. It currently fails silently.

Reply:
This one is iterating over attributes that are in spanData not in expected attributes, that's why didn't add fail condition. Every test case is not testing all attributes inside span.

Reply:
The common approach we take is to test all of what is expected. Without verifying that things things that shouldn't be included are not included you allow failure modes where additional, unintended, attributes can be added without being caught by the unit tests. The idea in having these tests is to ensure all predictable failure modes are avoided.

Can we fix the expected attributes to be comprehensive and test nothing more than what is intended is included?